### PR TITLE
fix: drop in default configs and use env vars for config

### PR DESCRIPTION
**Pull request by michaeldelago**
With the merging of https://github.com/azerothcore/azerothcore-wotlk/pull/16817 and https://github.com/azerothcore/azerothcore-wotlk/pull/17040, AzerothCore now supports the use of environment variables for configuration (primarily used for docker configuration). This negates the need for the "dockerdist" files and overall allows for a cleaner interface for applying dynamic configurations to AzerothCore.

To ensure that acore-docker is functioning, This PR adds copies in the ".conf.dist" configuration files to the `conf` directory to use as a base, and the environment variables from the upstream repo's compose file have been added to this repos docker-compose.yaml

### CODE REVIEW

**Comment by michaeldelago** [comment]
CI failed because it doesn't have the config hooked up. Seeing what can be done for this

**Comment by Helias** [comment]
`To ensure that acore-docker is functioning, This PR adds copies in the ".conf.dist" configuration files to the conf directory to use as a base, and the environment variables from the upstream repo's compose file have been added to this repos docker-compose.yaml`

This means that we have to maintain the *.conf here and on azerothcore-wotlk, is there a way to copy them from azerothcore in a linked way so we do not need to update the conf here and there at the same time?

**Comment by michaeldelago** [comment]
> This means that we have to maintain the *.conf here and on azerothcore-wotlk

@Helias 

The cleanest way I can think of would be to create a github action that runs on a schedule, which runs a curl command to each `conf.dist` and saves them to the appropriate path, and then commits those.

That can be separate from this PR though, I'm just trying to ensure this is working :D

In all honesty, I'm in favor of archiving this repository and in its place having better documentation and a better example docker-compose.yml in the upstream repo.

**Comment by michaeldelago** [comment]
> The cleanest way I can think of would be to create a github action that runs on a schedule, which runs a curl command to each conf.dist and saves them to the appropriate path, and then commits those.

Oh also if there's branch protections, a Deploy Key can be used to force push. It's better to do that than use someone's PAT or a separate account

**Comment by gamezonemalta** [comment]
guys, i'm new to this scenario and this pack, but i was wondering if there could be a solution for making the database in local mode only and also maybe change the passwords of such database? and i dont wish to use a seperate user so i dont wish to use .env

**Comment by michaeldelago** [comment]
@gamezonemalta

I'm not sure if your comment is relevant to this issue, though you can set a custom database password on the docker-compose.yml file with the `DOCKER_DB_ROOT_PASSWORD` environment variable. 

Please note that this repository only serves as an example, you can definitely make the tweaks that you see fit.

**Comment by michaeldelago** [comment]
@Helias since https://github.com/azerothcore/azerothcore-wotlk/pull/17071 has been merged into the core, the conf files are not necessary to be included in this PR.

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
thank you for your PR!

File: docker-compose.yml [hunk 1]
```diff
@@ -51,12 +51,18 @@ services:
       - ${DOCKER_SOAP_EXTERNAL_PORT:-7878}:7878
     expose:
       - 8085
-    volumes:
```
Comment (michaeldelago) [inline docker-compose.yml:54]: woops, thats bad!

**Pull request merged**